src/os/filestore/FileStore.cc

```diff
@@ -718,6 +718,14 @@ int FileStore::statfs(struct store_statfs_t *buf0)
   }
   buf0->total = buf.f_blocks * buf.f_bsize;
   buf0->available = buf.f_bavail * buf.f_bsize;
+  // Adjust for writes pending in the journal
+  if (journal) {
+    uint64_t estimate = journal->get_journal_size_estimate();
```

Comment:
For case 2, even some writes pending in the journal, but we could not say that the data doesn't reach to (file system size not changed) or has been calculated by background filesystem(file system size not changed). So we could say that the buf0->available value isn't  accurate at some time.